### PR TITLE
Fixes #14843 - Improve rake config error handling

### PR DESCRIPTION
**Pull request by dLobatog**
Currently rake config doesn't give you any hint when you provide an
invalid setting. Instead, it should tell you what was wrong with the
input, and try to guide you to a valid one.

This commit provides error handling for wrong JSON values, and prints
what the error was if the setting could not be saved.

For example, you can test this with:

```
rake config -- -k ignored_interface_identifiers -v 'P"a": "b"}'
rake config -- -k ignored_interface_identifiers -v '{"a": "b"}'
rake config -- -k db_pending_migration -v 3
```

Before, it would not tell you what went wrong, now it does.

### CODE REVIEW

File: lib/tasks/config.rake [hunk 1]
```diff
@@ -100,19 +100,19 @@ BANNER
...
         value = @key_values[key]
         setting = Setting.find_by_name(key)
-        old_value = setting.value
         if value == :unset
           value = nil
         elsif complex_type?(setting.settings_type)
           setting.value = typecast_value(setting.settings_type, value)
         else
-          setting.parse_string_value(value)
+          if setting.parse_string_value(value)
```
Comment (iNecas) [inline lib/tasks/config.rake]: shouldn't there be something inside the if body?  If no, unless should be used intead

**Comment by dLobatog** [comment]
@iNecas Updated, thanks for the review!

**Comment by iNecas** [comment]
What's expected behavior when calling `bundle exec rake config -- -k ignored_interface_identifiers -v '{"a": "b"}'`? I expected an error message explaining wrong type and non-zero exit status. I'm seeing it acting as everything is ok, by the value has not actually changed.

**Comment by iNecas** [comment]
Edit: the error message is show, but compare the output, when json syntax error vs. wrong type:

```
$ rake config -- -k ignored_interface_identifiers -v '"a": "b"}'
ERROR: Could not parse value "a": "b"} as JSON. Please check the value is a valid JSON array.
$ echo $?
2

$ bundle exec rake config -- -k ignored_interface_identifiers -v '{"a": "b"}'
ERROR: Invalid value {"a"=>"b"} for ignored_interface_identifiers - ["Value must be comma separated"]
ignored_interface_identifiers: {"a":"b"}
$ echo $?
0
```

I would expect that for validation error, it would not print out the `ignored_interface_identifiers: {"a":"b"}` line and the exit code would be non-zero as well

**Comment by dLobatog** [comment]
@iNecas Thanks for testing! It should return the appropriate error code and not print out the invalid setting now.

**Comment by iNecas** [comment]
[test] intermittent issues

**Comment by iNecas** [comment]
Thanks, works fine now. I will merge APJ

**Comment by dLobatog** [comment]
I don't think the issues have to do with this PR, the last run had 1 failure and 2 killed processes

**Comment by iNecas** [comment]
I wanted to give it another try, but my patience is over now:) https://github.com/theforeman/foreman/commit/5b53236fcc3acf0af29a2ddb60d17590d8db757f , Thanks @dLobatog

**Pull request closed without merging**